pkg/types/imagedata.go

```diff
@@ -0,0 +1,39 @@
+package types
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/anchore/stereoscope/pkg/file"
+	"github.com/anchore/stereoscope/pkg/image"
+)
+
+type ImageData struct {
+	*image.Image
+	FileMap map[string]FileData
+}
+
+type FileData struct {
+	RealPath file.Path
+	Content  []byte
+	FileMode os.FileMode
+}
+
+type FilterFunc func(filePath string, fileMode os.FileMode) (bool, error)
+
+func (f *FileData) ReadContentAndClose(img *image.Image) ([]byte, error) {
```

Comment:
I feel like it's just disclosing internals. maybe better just call it ReadContent? (also the call error print :) )

Reply:
can you please update also the call error prints (e.g. Failed to read content)?